### ISSUE TITLE
TimerTrigger does not read value from portal settings using the %setting_name%-convention

### ISSUE THREAD

**Issue by peter-bannerflow** [Azure/azure-functions-host#2461]
#### Investigative information

- Timestamp: 2018-02-23T20:02:00.0054864+00:00
- Invocation ID: 1ca4272c-b3a0-44c8-8aa2-b4833f1a8578
- Region: EU-North

Platform information:
.NET Core 2.0

Package information:
```xml
    <PackageReference Include="Microsoft.Azure.WebJobs" Version="3.0.0-beta3" />
    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions" Version="3.0.0-beta3" />
    <PackageReference Include="Microsoft.Extensions.Configuration" Version="2.0.0" />
    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="2.0.0" />
    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="2.0.0" />
    <PackageReference Include="Microsoft.Extensions.Options.ConfigurationExtensions" Version="2.0.0" />
    <PackageReference Include="Microsoft.Web.WebJobs.Publish" Version="1.1.0" />
```

#### Repro steps

Provide the steps required to reproduce the problem:

1. Implement a time triggered function that gets its schedule from appsettings.json
2. Deploy web job to Azure app service
3. Set a new value for the schedule in Azure portal under Application Settings. Example: `Settings:Schedule = 0 */10 * * * *`

#### Expected behavior

 - After I perform step 3 I expect the time triggered function to use the schedule that I have defined in the portal settings.

#### Actual behavior

- The function still use the value that is configured in the appsettings.json file. 

#### Related information 

<details>
<summary>Functions.cs</summary>

```csharp
        public async Task TimeTriggeredFunction(
            [TimerTrigger("%Settings:Schedule%")] TimerInfo timerInfo, 
            TraceWriter log)
        {

        }
```
</details>

<details>
<summary>appsettings.json</summary>

```json
{
  "ConnectionStrings": {
    "AzureWebJobsDashboard": "UseDevelopmentStorage=true",
    "AzureWebJobsStorage": "UseDevelopmentStorage=true"
  },
  "Settings": {
    "Schedule": "0 */2 * * * *"
  }
}
```
</details>

**Comment by pragnagopa** [comment]
This issue should have been fixed in v3.0.0-beta4. Here is the related [commit](https://github.com/Azure/azure-webjobs-sdk/commit/24447418dcfd5657e320937c7c7ce5d305c0f28c). Can you please try the latest nuger package?

**Comment by peter-bannerflow** [comment]
Thanks, updating to v3.0.0-beta4 fixed the issue with reading app settings. However, there seem to be a similar issue for connection strings. 

If I do:
```
var configuration = new ConfigurationBuilder()
   .SetBasePath(Directory.GetCurrentDirectory())
   .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
   .Build();

configuration["ConnectionStrings:SomeConnectionString"]
//OR
configuration.GetConnectionString("SomeConnectionString");
```

I always seem to get the value from the appsettings.json file and not the value that I have specified in the portal.
